### PR TITLE
simplify SDLC and minor code cleanups

### LINKED ISSUE #408: Feature request: Support typeahead to search for project in the homepage
## 🚀 Feature Request

If a user has a lot of projects, it will take a very long time to get all of them on the home page. We should allow typeahead.

### Motivation

Refer to how we do project dependency typeahead. There are several requirements:
- [ ] When `search string length <= 3` make a normal search with `limit = 10`
- [ ] Always make a `limit = 10` call
- [ ] `NOTE` this change will probably break some tests in Studio

### LINKED ISSUE #879: Feature request: Remove prototype project support
### Similar requests

- [X] I have searched and found no existing similar requests

### How are you using Studio?

_No response_

### What problems are you trying to solve?

The concept of prototype projects was devised in the earliest day of the platform to accommodate for playgrounds, proof-of-concept. They cause code branching in both https://github.com/finos/legend-sdlc and https://github.com/finos/legend-studio

In SDLC, prototype projects and production projects must come from separate underlaying VCS instances. We could and should simplify the code by removing this distinction, and instead, having configuration per SDLC instance.

The code path in SDLC 

### Describe the solution you would like to see implemented

- [x] Refactor https://github.com/finos/legend-sdlc - https://github.com/finos/legend-sdlc/pull/383
- [x] Simplify fetching of projects
- [x] Simplify project creation dialog
- [x] https://github.com/finos/legend-studio/issues/408
- [x] Remove `TEMPORARY__useSDLCProductionProjectsOnly` and `TEMPORARY__useSDLCProductionProjectsOnly` flags
- [x] Deploy separate instances for prototype and production Gitlab for existing infrastructures

### Describe alternatives you have considered

_No response_

### Documentation, Design, Adoption, Migration Strategy

_No response_

### Contribution

- [ ] I would like to work on this feature

### LINKED ISSUE #935: Feature request: Remove support for multiple SDLC instances
### Similar requests

- [X] I have searched and found no existing similar requests

### How are you using Studio?

Legend Studio

### What problems are you trying to solve?

We decided that the overall use case for multiple SDLC instances have a few drawbacks:
1. It's generally confusing in the UI to indicate this, especially it obfuscates the URL. Generally, we are quite conservative about URL patterns in our app
2. The code logic to support this is unnecessarily complicated.
3.  **BONUS** If we remove URL patterns setup used to support this feature, #688 would be much easier.

Hence, we decided to remove the support for it. We basically undo the work of #642 and #149

#### Tasks

- [x] Remove the code logic
- [x] Write docs about this (potentially)

#### For migration/backward compatibility

- [x] Either modularize the `NotFound` page behavior so for existing infrastructure or we could rely on the doc config provided to make decision whether or not the show the `NotFound` page
- [x] For existing infrastructure that deploys Studio with multiple SDLC instances, add new deployment of /studio at routes like `/gcp-cloud-sdlc-studio/`, `/onprem-sdlc-studio/`, `aws-sdlc-studio/`, etc. We can also make use of the existing `studio.config.js` file with tweak to change `baseUrl` based on an environment variable to create specific deployment artifacts
- [x] Refactor the path for `viewer mode with GAV` to `view/archive/...`

#### Extra 

- [x] rework `taxonomy` app support for multiple taxonomy tree support in https://github.com/finos/legend-studio/pull/692 i.e. `/taxonomy/tree/.../...`

### Describe alternatives you have considered

_No response_

### Documentation, Design, Adoption, Migration Strategy

_No response_

### Contribution

- [X] I would like to work on this feature

### PR DESCRIPTION
**Pull request by akphi**
## Summary

- [x] Closes #879
- [x] Closes #935 
- [x] Minor code smell cleanups (detected by `Sonar`)

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

#### New homepage and project creation modal dialog
See the documentation in markdown and where we lay them out

https://user-images.githubusercontent.com/13574879/163409350-27c69e24-97e8-4646-9cf1-a5870728da1b.mov

#### Not found page when documentation is provided and not provided

https://user-images.githubusercontent.com/13574879/162837158-ddb486ff-6b3d-484c-867d-c8e7a420c29d.mov

https://user-images.githubusercontent.com/13574879/162837159-d88d87ff-3995-41e7-9a73-0e971d1ff83c.mov

### CODE REVIEW

File: packages/legend-studio/src/stores/LegendStudioRouter.ts [hunk 1]
```diff
@@ -32,57 +30,23 @@ export enum LEGEND_STUDIO_PATH_PARAM_TOKEN {
...
-  EDIT_GROUP: generateRoutePatternWithSDLCServerKey(
-    `/edit/:${LEGEND_STUDIO_PATH_PARAM_TOKEN.PROJECT_ID}/groupWorkspace/:${LEGEND_STUDIO_PATH_PARAM_TOKEN.GROUP_WORKSPACE_ID}/`,
-  ),
-  SETUP: generateRoutePatternWithSDLCServerKey(
-    `/setup/:${LEGEND_STUDIO_PATH_PARAM_TOKEN.PROJECT_ID}?/:${LEGEND_STUDIO_PATH_PARAM_TOKEN.WORKSPACE_ID}?`,
-  ),
-  SETUP_GROUP: generateRoutePatternWithSDLCServerKey(
-    `/setup/:${LEGEND_STUDIO_PATH_PARAM_TOKEN.PROJECT_ID}/groupWorkspace/:${LEGEND_STUDIO_PATH_PARAM_TOKEN.GROUP_WORKSPACE_ID}/`,
-  ),
+  NOT_FOUND: `/notfound`,
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/stores/LegendStudioRouter.ts:34]: As hopefully with this we stabalize our routes, might make sense to document this somewhere about what each url routes to.

File: packages/legend-studio/src/components/LegendStudioApplication.tsx [hunk 2]
```diff
@@ -123,12 +184,23 @@ export const LegendStudioApplicationRoot = observer(() => {
               component={entry.component as React.ComponentType<unknown>}
             />
           ))}
-          <Redirect
-            to={generateSetupRoute(
-              applicationStore.config.defaultSDLCServerOption,
-              undefined,
-            )}
-          />
+          {/* <Route
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/components/LegendStudioApplication.tsx:187]: remove this please

File: packages/legend-studio/src/components/setup/Setup.tsx [hunk 3]
```diff
@@ -218,61 +208,78 @@ const CreateProjectModal = observer(() => {
...
-                  Project Type
-                </div>
-                <div className="panel__content__form__section__header__prompt">
-                  Any project intended for use in production must be a
-                  production project. Prototype projects should only be used as
-                  playgrounds or proofs-of-concept.
+            {allowCreatingNewProject && (
+              <div className="panel__content__form">
+                <div className="panel__content__form__section">
+                  <div className="panel__content__form__section__header__label">
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/components/setup/Setup.tsx:214]: Why are we using the word "import" here even when creating projects are allowed. 
Doesn't the flow make sense if
canCreateProject
  true: default to create, but allow import as another option
  false: just show import option?

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]
Can we add some tests after we go live with this please ? Thanks

File: packages/legend-application/src/stores/ApplicationStore.ts [hunk 4]
```diff
@@ -303,24 +303,6 @@ export class ApplicationStore<T extends LegendApplicationConfig> {
       });
       return;
     }
-    // See https://hackernoon.com/copying-text-to-clipboard-with-javascript-df4d4988697f
-    if (document.queryCommandSupported('copy')) {
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-application/src/stores/ApplicationStore.ts:307]: it's deprecated https://developer.mozilla.org/en-US/docs/Web/API/Document/queryCommandSupported, so is `execCommand`

File: packages/legend-extension-dsl-diagram/src/DiagramRenderer.ts [hunk 5]
```diff
@@ -2620,7 +2620,6 @@ export class DiagramRenderer {
 
       switch (this.interactionMode) {
         case DIAGRAM_INTERACTION_MODE.PAN: {
-          e.returnValue = false;
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-extension-dsl-diagram/src/DiagramRenderer.ts:2623]: deprecation notice again, check `Sonar` https://sonarcloud.io/project/overview?id=legend-studio

File: packages/legend-studio/src/components/editor/edit-panel/service-editor/__tests__/ServiceRegistrationTest.test.tsx [hunk 6]
```diff
@@ -123,15 +123,14 @@ const setup = async (
 
 test(
   integrationTest(
-    'Service Editor basic registration functionality for PRODUCTION projects',
+    'Service Editor basic registration functionality for projects with versions',
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-studio/src/components/editor/edit-panel/service-editor/__tests__/ServiceRegistrationTest.test.tsx:126]: I'll check again later

File: packages/legend-art/src/components/markdown/MarkdownTextViewer.tsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+export const MarkdownTextViewer: React.FC<{
+  value: MarkdownText;
+}> = (props) => {
+  const { value } = props;
+  const markdownHTMLText = renderMarkdownToHTML(value.value);
+
+  return (
+    <div
+      className="markdown-content"
+      dangerouslySetInnerHTML={{
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-art/src/components/markdown/MarkdownTextViewer.tsx:28]: Makes sense. Should use `react-markdown` instead. Somehow `markdown-it` stucks in my mind 😆

File: packages/legend-taxonomy/src/stores/LegendTaxonomyStore.ts [hunk 8]
```diff
@@ -319,7 +319,7 @@ export class TaxonomyNodeViewerState {
             this.taxonomyStore.applicationStore.navigator.openNewWindow(
               `${
                 this.taxonomyStore.applicationStore.config.studioUrl
-              }/view/${generateGAVCoordinates(
+              }/view/archive/${generateGAVCoordinates(
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-taxonomy/src/stores/LegendTaxonomyStore.ts:322]: Ukie

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**